### DISCUSSION TITLE
How to delete collection view?

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#2362] [Q&A]
When I try to delete a collection view it tells me: `Failed to delete collection due to existing dependency`.

**Comment by ganigeorgiev** [comment]
Is there a relation field pointing to the view?

**Reply by dschissler** [reply]
I'll check, but I don't think so.

**Reply by ganigeorgiev** [reply]
If possible, could you export your collections configuration as JSON (_Admin UI > Settings > Export collections_)?

**Reply by dschissler** [reply]
@ganigeorgiev I think that I figured it out.  1) Create a base collection.  2) Create a view from that base. 3) Rename the base collection, 4) Try to delete the view.

It was necessary for me to neuter the SQL in each view so that it didn't access invalid base collections before I could delete ANY of those views.  The checks are too restrictive and could put someone's system into a state where they have to shit can it because it's all locked up.

**Reply by ganigeorgiev** [reply]
Better error reporting and better automated collection changes sync (eg. in case of collection or field rename) is in my todo but for now remains a low priority since there are too many edge cases and requires more in depth research (_there are also some minor inconsistencies between the errors in the cgo and noncgo sqlite drivers that further complicate the issue_).

**Reply by dschissler** [reply]
Is there a way to get a force delete of a view?  Can base collections link to view collections?  If a view is invalid then perhaps there could just be a way of ending the madness?

**Reply by ganigeorgiev** [reply]
> Is there a way to get a force delete of a view?

Only programmatically.

> Can base collections link to view collections?

Yes. You can have a `relation` field pointing to a view or even one view with a relation from another (_file fields are also "proxied" up to 5 levels_).

**Reply by dschissler** [reply]
At any rate, this thread now exists.

**Comment by gaganbiswas** [comment]
Hey, I am getting the same problem but with a base collection. Isn't being referenced anywhere else. Just created it.

**Reply by ganigeorgiev** [reply]
@gaganbiswas You can start the executable with `--dev` and/or check _Dashboard > Logs_ for more detailed error messages. If you have custom hooks code, make sure that it is not caused by it. Other than this I don't know how to help further.

If you are still not able to resolve it, create a new Q&A discussion but export your collections configuration as json so that others can try to reproduce it.
